### DISCUSSION TITLE
Alien(s) upvent when player is inaccessible in some locations but not others

### DISCUSSION THREAD

**Discussion by ghramsey** [MattFiler/LegendPlugin#108] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/51859157/210480843-d06791be-2320-430f-ada9-ca2fa4e444e9.png)

I opened the transit office in mission 3.  Tech_Hub map,  The door normally needs the door code obtained
in mission 10 from the civilians.  I set it to have a handle instead and moved the save station from the transit 
platform inside the room.  I have not changed the door parameters except to edit the door mechanism enum
 to 3 which is a handle.
 
 Here's a screenshot of the door in question: 
![image](https://user-images.githubusercontent.com/51859157/210480339-9855542a-1650-4f14-a126-3a48825f82ee.png)
![image](https://user-images.githubusercontent.com/51859157/210480407-3d62827e-cc03-4389-91d4-c59bf9b8c076.png)

The aliens cannot trigger the door from the outside however instead of their normal behavior of seeking a vent when 
I'm in an inaccessible room or location they're camping and screaming at their inability.

I'm curious what is unique about this door and other locations that the aliens do not behave as they normally do
where the player cannot be reached?

I've experienced this phenomenon in the tech hub workshop where you get the SAT chip. 
The aliens cannot get you behind the record stacks. There seems to be a collision barrier to them only.
![image](https://user-images.githubusercontent.com/51859157/210481641-9e9ebd9d-3d59-4658-bb57-f8eab07adaf2.png)

In mission 7 in the warehouse there is a collision barrier that the aliens cannot pass between the first row of crates.  
And in mission 18 when one inside the vents where the aliens cannot go, they camp the exits instead of upventing.

It's as creepy from the screaming as it is frustrating b/c the alien AI is not behaving as it does in other similar conditions.

Any insights into this strange behavior will be helpful.

**Comment by MattFiler** [comment]
Is this not just regular behaviour when the Alien is trying to reach a victim that it can see via its viewcone, but not path to via the navmesh? I feel like I've seen it before. 

I think the only time the Alien will fix himself in relation to being stuck is if he's locked in a room that has no path to a vent, in which case I think he is hardcoded to teleport out. Might be wrong on that though.

**Reply by MattFiler** [reply]
Specifically in all of these cases, you're relying on dynamic CollisionBarriers to block the Alien's path. Perhaps it's exposing an issue with the game's programming - or perhaps you also need NavMeshBarrier entities in these places to fully tell the Alien that he cannot path there. Not too sure, but good spots!

**Reply by ghramsey** [reply]
So the specific activity I'm expecting is as in mission 7 the "safety vent" that is often exploited by speedrunners.
Make noise to lure the alien.  Open the vent as he's approaching and if he's in view he stops short, and does a 180.
If not in view you can hear him halt his charge and upvent.  He goes into a lock-on animation and can be made contact with. 
There are  lockable rooms in mission 4 where you get the noisemaker and mission 5 the rooms on Dr. Morley's rounds all 
have no vents to enter through so once the door is locked player is inaccessible.

Making noise when the alien is searching on the ground whether he sees you or not results in his seeking a vent immediately as an animation.  Again a speedrunner exploit often used when there are multiple aliens.

There seems to be nothing outwardly unique or odd about the other locations except the aliens' reactions being wholly different.
Perhaps something coded or otherwise not coded for pathing between these examples.

A NavMeshBarrier seems likely given the data.  I'm afraid I'm searching for a needle in a haystack in a pitchblack room though.
Any ideas where to search for the differences?

In Tech Comms>level_composities>tech_comms>environment_comms I see a NavMeshReachabilitySeedPoint_1 and coordinates.  The attached resources button bring up a window that has no data.  (Is this a bug ? )

One thing I did note on that door is it's Archetype is 2.
What does that parameter do   ?  I've never seen that on a door before now.
Perhaps it's significant.   The only archetype in your wiki is an ENUM Evironment_Archetype.
There's no parameter named that.

In the door_package archetype there's two settings for pathfindingwaitnode
seems to be a parameter called 'extra_cost ' set to 1.  This is not listed in your
extensive list of parameters and nodes.  Any idea what it does ?
Clicking on pathfindingnode and some of the links don't follow through to display.
Is this a bug? I think I see something similar already.

**Comment by ghramsey** [comment]
Looking at the animation and how the aliens react it looks like they're awaiting the door to open and it's not doing so.

What function allows or disallows the alien to trigger a door?  With respect to the transit office door I think whereever 
that trigger is also lies the answer to why they're not upventing when I'm not acessible.

**Reply by ghramsey** [reply]
![image](https://user-images.githubusercontent.com/51859157/210913364-104a9d6a-e60c-48ad-8dbb-c2b78a6b7946.png)

I found a piece of the puzzle.  As above message suggested I was thinking along the lines of whatever allows or stops the 
aliens  triggering the doors is probably at least part of this puzzle.  I'm not sure if they're same or if one follows the other.
But in my searching I found this in the door package function.  The Tech_Comms  door package is on the right the Tech_Hub on the left.
I swapped in that character_class_combination from 733 to 1023 and it had opposite effect; it enabled the door to be triggered by the aliens..   Since I'd asked about that ENUM before I guess I'm going to need to start experiementing w/ editing the value on this door package function.

**Comment by ghramsey** [comment, marked as answer]
I figured out setting the door to close and stopping the alien from triggering the door mysteriously does not inform the AI not to wait for it to open. However, using Force_close via a door button or other method sends this signal and the alien turns tail and upvents. Still curious activity.  In the techsupport area I'm curious why the game designers placed barriers like they did with no active alien in M3 until the end in the base game and no reason for a player to wander in there during later missions.  Headscratcher but it lends well to extra aliens by serrendipity. Lets one place a handy save spot. :)
